code/game/objects/items/RPD.dm

```diff
@@ -561,20 +561,27 @@ GLOBAL_LIST_INIT(RPD_recipes, list(
 	. = FALSE
 	switch(p_class) //if we've gotten this var, the target is valid
 		if(PAINT_MODE) //Paint pipes
+			if(!is_paintable)
+				return ..()
 			var/obj/machinery/atmospherics/pipe/P = A
-			playsound(get_turf(src), 'sound/machines/click.ogg', 50, 1)
-			P.paint(paint_colors[paint_color])
-			user.visible_message("<span class='notice'>[user] paints \the [P] [paint_color].</span>","<span class='notice'>You paint \the [P] [paint_color].</span>")
+			if(P)
+				playsound(get_turf(src), 'sound/machines/click.ogg', 50, 1)
+				P.paint(paint_colors[paint_color])
+				user.visible_message("<span class='notice'>[user] paints \the [P] [paint_color].</span>","<span class='notice'>You paint \the [P] [paint_color].</span>")
 			return
 
 		if(EATING_MODE) //Eating pipes
+			if(!is_consumable)
```

Comment:
do the same here with `is_consumable` and remove the check above that checks all three vars - it's redundant now that you've added these